samples/cube.lobster

```diff
@@ -0,0 +1,111 @@
+// This example renders a cube with a different colorèd lobster on each face.
+// Moving the mouse or WASD rotates the viewport around the cube like turning a
+// globe in front of the camera.
+//
+// The example illustrates creating a simple mesh and rendering a texture on
+// the mesh in various orientations.
+// The example showcases the render_to_texture utility, which allows us to
+// render a 2D drawing on the surfaces of a 3D object, using a frame buffer.
+//
+// Kris Kowal <kris@cixar.com>
+
+import vec
+import color
+import util3d
+
+fatal(gl_window("Lobster Cube", 515, 515))
+
+// Generate a mesh for a unit square.
+let square_corners = [ xy_0, xy_x, xy_1, xy_y, xy_0 ]
+let square_mesh = gl_new_poly(square_corners)
+
+// How to break glass when the texture coordinates do not match the mesh
+// vertices:
+// let square_positions = map(square_corners) v: xyz(v, 0)
+// let square_texcoords1 = square_corners
+// let square_indices = [0, 1, 2, 2, 3, 0]
+// let square_mesh = gl_new_mesh(
+//     "PT",
+//     square_positions,
+//     [], // square_colors,
+//     [], // square_normals,
+//     square_texcoords1,
+//     [], // square_textcoords2,
+//     square_indices
+// )
+
+let lobster_tex = gl_load_texture("data/textures/lobster.jpg")
+assert lobster_tex
+
+let detail = 256
+let face_tex = map([
+    color_purple,
+    color_olive,
+    color_teal,
+    color_dark_red,
+    color_dark_blue,
+    color_dark_green,
+]) color:
+    render_to_texture(nil, xy_1i * detail, false, nil, 0):
```

Comment:
while it is nice to demo this functionality, this is a really heavy-weight way to do coloring.. instead you can simply call `gl_color` in the `for(6)` loop below..

Reply:
I think to rationalize the demo, I’ll render the face number instead of the lobster, and also move the color out. I didn’t realize the color mask would thread through textures on meshes!

Reply:
Pushed this, so it renders numbers instead of lobsters.